lib/pacemaker/pcmk_fence.c

```diff
@@ -190,32 +190,34 @@ pcmk__fence_history(pcmk__output_t *out, stonith_t *st, char *target,
                            safe_str_eq(target, "*") ? NULL : target,
                            &history, timeout/1000);
 
-    out->begin_list(out, "event", "events", "Fencing history");
+    if (!cleanup) { // Cleanup doesn't return a history list
```

Comment:
I'd prefer to see this written more like:
```
if (cleanup) {
    stonith_history_free(history);
    return pcmk_legacy2rc(rc);
}
```

Or whatever.  And then the rest is indented normally.

Reply:
makes sense